### ISSUE TITLE
Windows failure if install directory has a "!" in it

### ISSUE THREAD

**Issue by jakcharvat** [flutter/flutter#26571]
I set up flutter and running any flutter command, including `flutter doctor` results in 

```
Error: The Flutter directory is not a clone of the GitHub project.
       The flutter tool requires Git in order to operate properly;
       to set up Flutter, run the following command:
       git clone -b beta https://github.com/flutter/flutter.git
```
Git is installed and running `git --version` returns `git version 2.20.1.windows.1`. 

I've tried installing Flutter through the ZIP on [flutter.io](https://flutter.io), as well as cloning it through both 
`git clone -b beta https://github.com/flutter/flutter.git`
and
`git clone -b stable https://github.com/flutter/flutter.git`,

restarted my laptop after each and still get the same error after running `flutter doctor`. 

I've never had this happen to me before, and would greatly appreciate any help. Thank you

**Comment by jakcharvat** [comment]
Never mind, it seems it was an issue with having an exclamation mark in the path to the sdk (`C:\Users\jakch\dev\flutter\!sdk\flutter\bin`) and removing it fixed the issue. It might be helpful to provide a clearer error message in this case though, as the problem was completely unrelated to the github repo.

**Comment by kangwang1988** [comment]
It seems a bug in flutter.bat
I will have a look.

**Comment by TheRedBull** [comment]
I ran into this issue today - I hold my dev tools in a directory called "!DevTools". After finding this issue, I moved the flutter folder from that directory to another (and changed the PATH var) which worked as expected.

It appears that the "SETLOCAL ENABLEDELAYEDEXPANSION" line in the /bin/.bat files is the main offender here (per various StackOverflow and google searches) - I changed that line to "SETLOCAL EnableExtensions DisableDelayedExpansion" (per the same research) and the flutter commands worked as expected when I moved everything back to the original folder

However, there is a like issue in the /bin/internal/update_dart_sdk.ps1 - which is choking on the ! in the directory name. I'm not well versed in PS scripting so I will continue my research but I wanted to bump this topic

**Comment by warp929** [comment]
I just ran into this issue in Windows 10.  The parent folder to flutter had an ! in the name.  This cause "flutter doctor" to fail.  Command prompt would close without warning.  Powershell would give me some sort of Git error when running "flutter doctor".  Cloned again.  Moving "flutter" to a folder without an exclamation mark and changing the environment variable fixed the problem.  Now "flutter doctor" runs.

**Comment by moko256** [comment]
This is fixed by flutter/engine#106861. I'll close this issue.
